### PR TITLE
add general fee quest

### LINKED ISSUE #873: General Fee Quest
### General
_Currently, there are quests to complete whether there is a fee for toilets or a car parks.
But there are many other OSM objects where the `fee` tag could be used._

Affected key(s) (or tags) to be modified: [fee](https://wiki.openstreetmap.org/wiki/Key:fee)
Question asked: **Does it cost something to use this [object type]?**

### Checklist
Checklist for quest suggestions (see guidelines):
- [x] 🚧 To be added tag is established and has a useful purpose
- [x] 🤔 Any answer the user can give must have an equivalent tagging (Quest should not reappear to other users when solved by one)
- [x] 🐿️ Easily answerable by everyone from the outside but a survey is necessary 
- [x] 💤 Not an overwhelming percentage of elements have the same answer (No spam)
- [x] 🕓 Applies to a reasonable number of elements (Worth the effort)

### Ideas for implementation
**Element selection:** (see list of objects below) with `[!fee]`

**Metadata needed:** Not necessarily, but maybe some objects could be disabled in some countries if there is a strong tendency for things to be (not) free so that it is obsolete to ask it

**Proposed GUI:** Simply a yes/no form as it already exists for the 2 current fee quests.

### (Incomplete) List of POIs possibly with fee:
- Museums: `tourism=museum`
- Attractions: `tourism=attraction`
- Theme parks: `tourism=theme_park` (could be "no" if it is to be paid at the particular attractions)
- Galleries: `tourism=gallery`
- Viewpoints: `tourism=viewpoint`
- Public gardens: `leisure=garden`
- Beach resorts: `leisure=beach_resort`
- Charging stations: `amenity=charging_station`
- Newspaper vending "machines": `[amenity=vending_machine][vending=newspapers]`
- Charging stations (for mobile phones etc.): `amenity=device_charging_station`
- and of course `amenity=toilets` and `amenity=parking`

### PR DESCRIPTION
**Pull request by matkoniecz**
fixes #873

-----

My work on this pull request was sponsored by a NGI Zero Discovery [grant](https://www.openstreetmap.org/user/Mateusz%20Konieczny/diary/368849)

### CODE REVIEW

**Comment by HolgerJeromin** [comment]
Correct issue number is #873

File: app/src/main/java/de/westnordost/streetcomplete/quests/general_fee/AddGeneralFee.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.SimpleOverpassQuestType
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao
+import de.westnordost.streetcomplete.quests.YesNoQuestAnswerFragment
+
+class AddGeneralFee(o: OverpassMapDataDao) : SimpleOverpassQuestType<Boolean>(o) {
+
+    override val tagFilters = "nodes, ways, relations with (tourism=museum or leisure=beach_resort or tourism=gallery) and access !~ private|no and !fee and name"
```
Comment (HolgerJeromin) [inline app/src/main/java/de/westnordost/streetcomplete/quests/general_fee/AddGeneralFee.kt]: IIRC other quests have line breaks in it to get clean diffs in the future.

**Comment by matkoniecz** [comment]
Should be fixed now, thanks for a review!

**Comment by westnordost** [comment]
Looks good, thank you!

**Pull request merged**